app/src/main/cpp/skyline/services/settings/ISettingsServer.cpp

```diff
@@ -23,4 +24,14 @@ namespace skyline::service::settings {
         response.Push<i32>(constant::NewLanguageCodeListSize);
         return {};
     }
+
+    Result ISettingsServer::GetRegionCode(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        region::RegionCode regionCode{*state.settings->systemRegion};
+
+        if (regionCode == region::RegionCode::Auto)
+            regionCode = region::GetRegionCodeForSystemLanguage(*state.settings->systemLanguage);
+
+        response.Push<i32>(static_cast<i32>(regionCode));
```

Comment:
No need to cast here

Reply:
yes, you need to, it's either `response.Push<i32>(static_cast<i32>(regionCode))` or `response.Push(regionCode)`

Reply:
Yes the latter lol